src/Main.elm

```diff
@@ -1632,38 +1821,88 @@ viewFormFieldOptionsPreview config fieldID formField =
                     )
                 ]
 
-        ChooseMultiple choices ->
+        ChooseMultiple { choices, minRequired, maxAllowed } ->
             let
                 values =
                     Dict.get fieldName config.trackedFormValues
                         |> Maybe.withDefault []
+
+                selectedCount =
+                    List.length values
+
+                -- Create validation messages when constraints aren't met
+                -- Removed validationMessage since we're using CSS for validation indication
+                -- Determine if validation is satisfied
+                isValid =
+                    case ( minRequired, maxAllowed ) of
+                        ( Just min, Just max ) ->
+                            selectedCount >= min && selectedCount <= max
+
+                        ( Just min, Nothing ) ->
+                            selectedCount >= min
+
+                        ( Nothing, Just max ) ->
+                            selectedCount <= max
+
+                        ( Nothing, Nothing ) ->
+                            True
+
+                -- Add validation element for CollectData mode (just the hidden input for validation)
+                validationElement =
+                    -- Only apply validation in CollectData mode, not in Editor mode
+                    if config.needsFormLogic && (minRequired /= Nothing || maxAllowed /= Nothing) then
+                        [ input
+                            [ type_ "number"
+                            , required True
+                            , attribute "value" (String.fromInt selectedCount) -- raw value for browser only
+                            , Attr.min (Maybe.map String.fromInt minRequired |> Maybe.withDefault "")
+                            , Attr.max (Maybe.map String.fromInt maxAllowed |> Maybe.withDefault "")
+                            , class "tff-visually-hidden"
+                            ]
+                            []
+                        ]
+
+                    else
+                        []
```

Comment:
this hidden `input[number]` is what we use to prevent forms (that checks for fields validity) to know if we are valid or not (does number of checkboxes selected fall within the range of min/max)